swiftlang/swift
# add option for "tensorflow toolchain" without tensorflow-swift-apis

The only functionality change is adding a `--no-tensorflow-swift-apis` option to `utils/build-toolchain-tensorflow`, so that we can build a toolchain with the same settings as our normal tensorflow toolchains but without including the tensorflow-swift-apis library.

This PR includes a cleanup that simplifies the functionality change: I removed all tensorflow-related build-script options (`--enable-tensorflow`, `--enable-tensorflow-gpu`, `--enable-x10`, `--tensorflow-host-lib-dir`, `--tensorflow-host-include-dir`, `--tensorflow-target-lib-dir`, `--tensorflow-bazel-options`). This should not change any functionality because these flags already do not do anything.

This PR is careful to not change the API of `build-toolchain-tensorflow`, so that existing callers still work. (e.g. some automated builds specify `--gpu`, which already has no effect, so I made `build-toolchain-tensorflow` emit a warning message but proceed whenever it sees `--gpu`). We can update the callers and eventually remove the deprecated flags.

Tests that I will do before merging:
* Build a toolchain using our google-internal automated Linux build, and verify that it still includes the tensorflow-swift-apis library.
* Build a toolchain on my mac machine and verify that it still includes the tensorflow-swift-apis library.
